### DISCUSSION TITLE
AAPS follower

### DISCUSSION THREAD

**Discussion by psonnera** [NightscoutFoundation/xDrip#3517] [Ideas]
AAPS 3.2 provides a broadcast method to xDrip+ that is pretty neat to extend AAPS capacities. It could also be used to completely follow AAPS on another phone but it requires xDrip+ to be setup as Sync Follower hence cannot be master.
![image](https://github.com/NightscoutFoundation/xDrip/assets/33809361/8df9af7c-64ce-4909-9d19-12e20ea4d9bf)
I know it doesn't make sense for an xDrip+ Sync Follower to be master but in this case it would...
How complex would it be to allow a Sync Follower to be master? Like with an additional entry like "Make AAPS follower master" or, worst case duplicate xDrip+ Sync follower data source to AAPS follower allowing xDrip+ to be master in this case?
Just an idea like that.
![image](https://github.com/NightscoutFoundation/xDrip/assets/33809361/530dfbc8-dc8e-4415-9b58-fb484f3a476a)

**Comment by Navid200** [comment]
What app will be the CGM?
If xDrip is a sync follower of AAPS, where will AAPS get its blood glucose readings from?
Can you explain this for someone who doesn't know anything about it, like for me?

**Reply by psonnera** [reply]
When using AAPS with xDrip+ as the data source, this is not necessary as the mechanism is embedded in xDrip+. The use case would be for someone using AAPS getting data from another app (not xDrip+) and wanting to use xDrip+ to share all data without using Nightscout.
Now, thinking about it, it would be easy to implement without changing much (assuming it does make sense) as AAPS as a data source is already embedded in xDrip+:
![image](https://github.com/NightscoutFoundation/xDrip/assets/33809361/c68be2bd-48d6-4593-8925-3155e365d5e6)
At that point, if data source is xDrip+ Sync and the user selects master
![image](https://github.com/NightscoutFoundation/xDrip/assets/33809361/83b11013-fa49-48f3-aa9c-4c18c62b3b99)
That means this follower will broadcast to others.

Still, I'm afraid the mechanism underlying would require many modifications... might not be as easy as I think...

**Reply by Navid200** [reply]
I think we can add a data source to xDrip being AAPS.  But, I don't see how we can call that a *hardware* data source.
But, that is not an issue.
Can you give me an example of a glucose source that only sends data to AAPS but not to xDrip?

I know one can forget to enable broadcast to xDrip when setting up byoda.  But, that is just a matter of setting it up correctly a second time.

Is there a sensor that AAPS can collect from, but xDrip cannot?

**Reply by psonnera** [reply]
@Navid200 I think you're right. You can get Libre, Dexcom, etc... sensor data into AAPS and simultaneously into xDrip+ using several methods so it is not absolutely necessary.
Thanks!
